packages/components/src/utils/supportsNativePopover.ts

```diff
@@ -0,0 +1,6 @@
+/**
+ * Checks if the current environment supports the native Popover API.
+ *
+ * @returns {boolean} `true` if the native Popover API is supported, `false` otherwise.
+ */
+export const supportsNativePopover = (): boolean => HTMLElement.prototype.hasOwnProperty('popover');
```

Comment:
Do we need to call this over and over again? Why not assign the result to constant like we do for `isHighContrastMode` or `hasConstructableStylesheetSupport`.